src/components/tl/nccl/tl_nccl_team.c

```diff
@@ -69,15 +75,17 @@ ucc_status_t ucc_tl_nccl_team_destroy(ucc_base_team_t *tl_team)
 #if NCCL_USE_NON_BLOCKING
     ncclResult_t nccl_status, st;
 
-    if (team->nccl_comm && team->comm_state == UCC_INPROGRESS) {
+    if (team->comm_state == TL_NCCL_COMM_STATE_DESTROY_COMM) {
```

Comment:
can we merge this if with the one below ? Maybe avoid the goto?
I am not sure to fully understand the flow of this function.
Shouldn't we free the Cuda stream even if `team->comm_state == TL_NCCL_COMM_STATE_DESTROY_COMM` ?

Reply:
TL_NCCL_COMM_STATE_DESTROY_COMM means cleanup function was already called and currently inprogress. comm_destroy is nonblocking operation just like team create. 

>Shouldn't we free the Cuda stream even if team->comm_state == TL_NCCL_COMM_STATE_DESTROY_COMM

no, stream is already destroyed if state is TL_NCCL_COMM_STATE_DESTROY_COMM 

>can we merge this if with the one below ? Maybe avoid the goto?

goto is used to go to comm destroy progress